File: incident/src/main/java/at/ase/respond/incident/persistence/model/Location.java
package at.ase.respond.incident.persistence.model;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Represents a location value object containing coordinates and address.
 */
@Data
@Builder
@Embeddable
@NoArgsConstructor
@AllArgsConstructor
public class Location {

    private LocationCoordinates coordinates;

    private LocationAddress address;

}

File: dashboard/src/app/dto/location.ts
import {LocationAddress} from "./location-address";
import {LocationCoordinates} from "./location-coordinates";

/**
 * Represents a location containing a description as well as coordinates.
 *
 * @param address the description of this location
 * @param coordinates the coordinates of this location
 */
export interface Location {
  address: LocationAddress;
  coordinates: LocationCoordinates;
}
